libs/deal-ticket/src/components/time-in-force-selector.tsx

```diff
@@ -44,7 +45,13 @@ export const TimeInForceSelector = ({
             timeInForce === VegaWalletOrderTimeInForce.FOK ||
             timeInForce === VegaWalletOrderTimeInForce.IOC
         );
-
+  useEffect(() => {
```

Comment:
chore (blocking): I'd prefer not to use useEffect like this? Lets just change the TIF state in the function where order type changes.

Reply:
updated